hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/InodeTree.java

```diff
@@ -212,6 +225,33 @@ void addLink(final String pathComponent, final INodeLink<T> link)
       }
       children.put(pathComponent, link);
     }
+
+    void addDirLink(final String pathComponent, final INodeDirLink<T> dirLink) {
+      children.put(pathComponent, dirLink);
+    }
+  }
+
+  /**
+   * Internal class to represent nested mount points in INodeTree. Nested mount points are non-leaf nodes in INode tree.
+   * Hence it inherits INodeDir. But it has additional attributes so wrap it with link.
+   * @param <T>
+   */
+  static class INodeDirLink<T> extends INodeDir<T> {
```

Comment:
Can you have a comment on what are the semantics of nested mount points? Currently, it's implicit in the code and the unit tests - it would be good to have an explicit declaration of this.